### PR TITLE
Authentication for Mobile & Desktop Apps with PKCE

### PR DESCRIPTION
**Pull request by khorne3**

### CODE REVIEW

**Comment by khorne3** [comment]
@mpaktiti Do you mind reviewing this for accuracy?

File: articles/client-auth/v2/mobile-desktop.md [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+---
+
+# Authentication for Mobile & Desktop Apps
+
+You can authenticate users of your mobile/desktop applications by:
+
+* Using the [Lock](/libraries/lock) client libraries;
+* Calling the Auth0 OAuth 2.0 endpoints.
+
+This article will cover how to call the Auth0 OAuch 2.0 endpoints using [Proof Key for Code Exchange (PKCE)](/api-auth/grant/authorization-code-pkce) during the authentication process.
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: Typo `Auth0 OAuch 2.0`. Better to replace it with `[Auth0 Authentication API](/api/authentication) endpoints`.

File: articles/client-auth/v2/mobile-desktop.md [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+
+### Authorize the User
+
+Once you've created the `code_verifier` and the `code_challenge` that you include in the authorization request, you'll need to obtain the user's authorization. This is technically the beginning of the authorization flow, and this step may include one or more of the following processes:
+
+* Authenticating the user;
+* Redirecting the user to an Identity Provider to handle authentication;
+* Checking for active SSO sessions.
+
+To authorize the user, your application must send the user to the [authorization URL](/api/authentication#authorization-code-grant-pkce-) (which includes the `code_challenge` you generated in the previous step, as well as the method you used to generate the `code_challenge`). Your URL should following this format:
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: `Your URL should following this format:` -> `Your URL should follow this format:`

File: articles/client-auth/v2/mobile-desktop.md [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+
+The **Create Client** window will open, allowing you to enter the name of your new Client. Choose **Native** as the **Client Type**. When done, click on **Create** to proceed.
+
+::: panel-danger Warning
+The Authorization Code flow with PKCE can only be used for Native Clients.
+:::
+
+![](/media/articles/client-auth/mobile-desktop/create-client.png)
+
+Once Auth0 creates the Client, navigate to the Client's **Settings** tab to add the following to the **Allowed Callback URLs** field: `https://${account.namespace}/mobile`
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: The callback URL does not match the one you refer to here https://github.com/auth0/docs/pull/3754/files#diff-f25c8abb80f21f168c9477732b90a3ecR189

File: articles/client-auth/v2/mobile-desktop.md [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+```text
+HTTP/1.1 302 Found
+Location: https://YOUR_APP/callback?code=AUTHORIZATION_CODE
+```
+
+Note the authorization code included at the end of the included URL.
+
+### Obtain an ID Token
+
+Using the authorization code obtained in step 2, you can obtain the ID token by making the appropriate `POST` call to the [tokens endpoint](api/authentication#authorization-code-pkce-).
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: Since you refer to steps I would add numbers to these paragraphs:
- `1. Create a Random Key and the Code Challenge`
- `2. Authorize the User`
- `3. Obtain an ID Token`

File: articles/client-auth/v2/mobile-desktop.md [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+  "expires_in":86400
+}
+```
+
+Note that Auth0 includes the `refresh_token` only if you've:
+
+* Included the `offline_access` scope in your initial call to the `authorize` endpoint;
+* Enabled **Allow Offline Access** for your API in the using the [API Settings tab](${manage_url}/#/apis) in the Auth0 Dashboard.
+
+### The `id_token`
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: I would set this as `##`, not as part of the `Implement Authentication`

File: articles/client-auth/v2/mobile-desktop.md [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+| exp | The _expiration time_. A number representing a speciﬁc date and time in the format “seconds since epoch” as [deﬁned by POSIX6](https://en.wikipedia.org/wiki/Unix_time). This claim sets the exact moment from which this **JWT is considered invalid**.<br/><br/>**This is a [registered claim](https://tools.ietf.org/html/rfc7519#section-4.1) according to the JWT Specification** |
+| iat | The _issued at time_. A number representing a speciﬁc date and time (in the same format as `exp`) at which this **JWT was issued**.<br/><br/>**This is a [registered claim](https://tools.ietf.org/html/rfc7519#section-4.1) according to the JWT Specification** |
+
+The exact claims contained in the `id_token` will depend on the `scope` parameter you sent to the `/authorize` endpoint. An Auth0 `id_token` will always include the **registered claims** and the `sub` claim, but the others depends on the `scope`.
+
+### Keep the User Logged In
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: I would move this before `## The `id_token` paragraph (keep it as part of `Implement Authentication`)

File: articles/client-auth/v2/mobile-desktop.md [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+:::
+
+The payload's claims can include some or all of the following:
+
+| Parameter | Description |
+|:------------------|:---------|
+| name | The name of the user which is returned from the Identity Provider. |
+| email | The email address of the user which is returned from the Identity Provider. |
+| picture | The profile picture of the user which is returned from the Identity Provider. |
+| sub | The unique identifier of the user. This is guaranteed to be unique per user and will be in the format (identity provider)&#124;(unique id in the provider), e.g. github&#124;1234567890. |
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: Format change: This is guaranteed to be unique per user and will be in the format `(identity provider)&#124;(unique id in the provider)`, e.g. `github&#124;1234567890`.

File: articles/client-auth/v2/mobile-desktop.md [hunk 8]
```diff
@@ -0,0 +1,444 @@
...
+
+* Authenticating the user;
+* Redirecting the user to an Identity Provider to handle authentication;
+* Checking for active SSO sessions.
+
+To authorize the user, your application must send the user to the [authorization URL](/api/authentication#authorization-code-grant-pkce-) (which includes the `code_challenge` you generated in the previous step, as well as the method you used to generate the `code_challenge`). Your URL should following this format:
+
+```text
+https://${account.namespace}/authorize?
+    audience=API_AUDIENCE&
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: I would remove the `audience` param. It only makes sense if you want to access an API and this article is only about authenticating a user.

File: articles/client-auth/v2/mobile-desktop.md [hunk 9]
```diff
@@ -0,0 +1,444 @@
...
+{
+  "access_token":"eyJz93a...k4laUWw",
+  "refresh_token":"GEbRxBN...edjnXbL",
+  "id_token":"eyJ0XAi...4faeEoQ",
+  "token_type":"Bearer",
+  "expires_in":86400
+}
+```
+
+Note that Auth0 includes the `refresh_token` only if you've:
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: I would remove references to `Enable **Allow Offline Access** for your API`. We didn't tell them anywhere in this doc to create an API, this is only for user authentication, not API authorization.

I would add a note though that the `access_token` can be used to call the `/userinfo` endpoint of the Authentication API https://auth0.com/docs/api/authentication#get-user-info

File: articles/client-auth/v2/mobile-desktop.md [hunk 10]
```diff
@@ -0,0 +1,444 @@
...
+    client_id=${account.clientId}&
+    code_challenge=CODE_CHALLENGE&
+    code_challenge_method=S256&
+    redirect_uri=${account.callback}
+```
+
+Request Parameters:
+
+* `audience`: The unique identifier of the target API. Use the __Identifier__ value in [API Settings](${manage_url}/#/apis). If you don't see this page in the Dashboard, enable the __Enable APIs Section__ toggle at [Account Settings > Advanced](${manage_url}/#/account/advanced).
+* `scope`: The [scopes](/scopes) for which you want to request authorization. Each scope must be separated from the others using a whitespace character. You can request any of the [standard OIDC scopes](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims) for users (such as `profile` and `email`), custom claims conforming to a namespaced format (see note below on Arbitrary Claims), and any scopes supported by the target API. Include `offline_access` to get a refresh token (make sure that you've enabled __Allow Offline Access__ in your [API Settings](${manage_url}/#/apis)).
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: `openid` should be set (at least)

File: articles/client-auth/v2/mobile-desktop.md [hunk 11]
```diff
@@ -0,0 +1,444 @@
...
+
+### Keep the User Logged In
+
+Auth0 assists in authenticating a user, but your application must keep track of whether or not a user is logged in. You can keep a global variable or a singleton object inside your application to do this. You can also use this object to store information about the user (such as name and profile image) so that you can deliver a personalized user experience within your application.
+
+## Example Use Cases
+
+This section covers use cases that illustrate the authentication process using PKCE.
+
+### Basic Authentication Request
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: Remove this example, you explain it better in the previous paragraphs and this does not add to it.

File: articles/client-auth/v2/mobile-desktop.md [hunk 12]
```diff
@@ -0,0 +1,444 @@
...
+{
+  "name": "auth0user@...",
+  "picture": "https://example.com/profile-pic.png",
+  "iss": "https://auth0user.auth0.com/",
+  "sub": "auth0|581...",
+  "aud": "xvt...",
+  "exp": 1478113129,
+  "iat": 1478077129
+}
+```
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: Add after this the github example https://auth0.com/docs/client-auth/mobile-desktop#request-a-user-log-in-with-github Just use this request https://github.com/auth0/docs/pull/3754/files#diff-f25c8abb80f21f168c9477732b90a3ecR393 and add `connection=github`

**Review by mpaktiti — CHANGES_REQUESTED** [review]
Hey Katie, I left some comments. Let me know when you address them so I can have another look. Thanks!

**Comment by mpaktiti** [comment]
@khorne3 Forgot to mention in my review that the directory will be different, not `v2`, but don't do any changes related to that now. I'll let you know as soon as versioning works again.

**Comment by khorne3** [comment]
@mpaktiti Ready for re-review. Thanks!

File: articles/client-auth/v2/mobile-desktop.md [hunk 13]
```diff
@@ -0,0 +1,433 @@
...
+To authorize the user, your application must send the user to the [authorization URL](/api/authentication#authorization-code-grant-pkce-) (which includes the `code_challenge` you generated in the previous step, as well as the method you used to generate the `code_challenge`). Your URL should follow this format:
+
+```text
+https://${account.namespace}/authorize?
+    scope=SCOPE&
+    response_type=code&
+    client_id=${account.clientId}&
+    code_challenge=CODE_CHALLENGE&
+    code_challenge_method=S256&
+    redirect_uri=${account.callback}/mobile
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: This redirect_uri translates into -> `https://YOUR_APP/callback/mobile`
This is not the same with this https://github.com/auth0/docs/pull/3754/files#diff-f25c8abb80f21f168c9477732b90a3ecR40

Can you please cross check that the callback url is the same throughout the doc?

File: articles/client-auth/v2/mobile-desktop.md [hunk 14]
```diff
@@ -0,0 +1,433 @@
...
+```
+
+Request Parameters:
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: I would remove all these and point to https://auth0.com/docs/api-auth/tutorials/authorization-code-grant-pkce#3-get-the-user-s-authorization for details on the params.

File: articles/client-auth/v2/mobile-desktop.md [hunk 15]
```diff
@@ -0,0 +1,433 @@
...
+  redirect_uri=com.myclientapp://myclientapp.com/callback">
+  Sign In
+</a>
+```
+
+If all goes well, you'll receive an HTTP 302 response:
+
+```text
+HTTP/1.1 302 Found
+Location: https://YOUR_APP/callback?code=AUTHORIZATION_CODE
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: The callback URL is/redirect_uri is not consistent throughout the doc

File: articles/client-auth/v2/mobile-desktop.md [hunk 16]
```diff
@@ -0,0 +1,433 @@
...
+    { "name": "Content-Type", "value": "application/json" }
+  ],
+  "postData": {
+    "mimeType": "application/json",
+    "text": "{\"grant_type\":\"authorization_code\",\"client_id\": \"${account.clientId}\",\"code_verifier\": \"YOUR_GENERATED_CODE_VERIFIER\",\"code\": \"YOUR_AUTHORIZATION_CODE\",\"redirect_uri\": \"com.myclientapp://myclientapp.com/callback\", }"
+  }
+}
+```
+
+Request Parameters:
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: You can remove all these and redirect the user to this doc https://auth0.com/docs/api-auth/tutorials/authorization-code-grant-pkce#4-exchange-the-authorization-code-for-an-access-token for details on params

File: articles/client-auth/v2/mobile-desktop.md [hunk 17]
```diff
@@ -0,0 +1,433 @@
...
+
+:::panel-info Access Tokens
+You can use `access_token` to call the [Authentication API's `/userinfo` endpoint](/api/authentication#get-user-info).
+:::
+
+### Keep the User Logged In
+
+Auth0 assists in authenticating a user, but your application must keep track of whether or not a user is logged in. You can keep a global variable or a singleton object inside your application to do this. You can also use this object to store information about the user (such as name and profile image) so that you can deliver a personalized user experience within your application.
+
+## The `id_token`
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: I think this section belongs to the id-token doc https://auth0.com/docs/tokens/id-token Can you remove these and refer to the doc for details? Before you do that you can check if all this information is present in the id-token doc (I don't think so) and if not add them.

File: articles/client-auth/v2/mobile-desktop.md [hunk 18]
```diff
@@ -0,0 +1,433 @@
...
+https://${account.namespace}/authorize?
+    scope=openid%20name%20picture&
+    response_type=code&
+    client_id=${account.clientId}&
+    code_challenge=CODE_CHALLENGE&
+    code_challenge_method=S256&
+    redirect_uri=https://${account.namespace}/mobile
+```
+
+After the user provides submits their credentials, your app receives an HTTP 302 response with a URL containing the authorization code at the end: `https://YOUR_APP/callback?code=AUTHORIZATION_CODE`
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: Again the comment on redirect_url/callback url: In line 338 it's `redirect_uri=https://${account.namespace}/mobile` while in line 341 it's `https://YOUR_APP/callback?code=AUTHORIZATION_CODE`. It should be similar for example:
- line 338 `redirect_uri=https://YOUR_APP/callback`
- line 341 `https://YOUR_APP/callback?code=AUTHORIZATION_CODE`

**Review by mpaktiti — CHANGES_REQUESTED** [review]
Katie, I left some more comments.

**Comment by khorne3** [comment]
@mpaktiti I opted to replace all callback URIs with the mobile version (is that ok?) and moved a sizable chunk of the ID token info to the ID token page (which I'd like to review at a later point to clean it up).

File: articles/client-auth/v2/mobile-desktop.md [hunk 19]
```diff
@@ -0,0 +1,394 @@
...
+description: Explains how to authenticate users in a mobile or desktop application.
+toc: true
+---
+
+# Authentication for Mobile & Desktop Apps
+
+You can authenticate users of your mobile/desktop applications by:
+
+* Using the [Lock](/libraries/lock) client libraries;
+* Calling the Auth0 [Authentication API](/api/authentication) endpoints.
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: I would add something to clarify the difference between these two options (for the sake of those not familiar with out product). Something simple like: 

```
* Using [Lock](/libraries/lock), a drop-in authentication widget that provides a standard set of behaviors and a customizable user interface, or
* Calling the Auth0 [Authentication API](/api/authentication) endpoints, in case you want to implement your own UI.
```

File: articles/client-auth/v2/mobile-desktop.md [hunk 20]
```diff
@@ -0,0 +1,394 @@
...
+
+This article will cover how to call the Auth0 [Authentication API](/api/authentication) endpoints using [Proof Key for Code Exchange (PKCE)](/api-auth/grant/authorization-code-pkce) during the authentication process.
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: Maybe link to the alternative? Something like: `If you are interested in implementing this with Lock, refer to [Lock v2 for iOS](/libraries/lock-ios) or [Lock for Android](/libraries/lock-android)`.

File: articles/client-auth/v2/mobile-desktop.md (hunk 19 above)
Comment (mpaktiti) [reply articles/client-auth/v2/mobile-desktop.md]: By the way, isn't auth0.js a valid option? Maybe follow up with Jeff or crew-brucke.

File: articles/client-auth/v2/mobile-desktop.md [hunk 21]
```diff
@@ -0,0 +1,394 @@
...
+
+## Overview
+
+Auth0 exposes OAuth 2.0 endpoints that you can use to authenticate users. You can call these endpoints through an embedded browser in your **native** application. After authentication completes, you can return an `id_token` that contains the user's profile information.
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: I would replace `id_token` with a link at the doc.

File: articles/client-auth/v2/mobile-desktop.md [hunk 22]
```diff
@@ -0,0 +1,394 @@
...
+
+Once Auth0 creates the Client, navigate to the Client's **Settings** tab to add the following to the **Allowed Callback URLs** field: `https://${account.namespace}/mobile`
+
+Scroll to the bottom of the page and click **Save**.
+
+![](/media/articles/client-auth/mobile-desktop/allowed-callback-url.png)
+
+## Implement Authentication
+
+To implement the Authorization Code Grant Flow using Proof Key for Code Exchange, you will need to execute the following steps:
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: Link to the doc https://auth0.com/docs/api-auth/grant/authorization-code-pkce

File: articles/client-auth/v2/mobile-desktop.md [hunk 23]
```diff
@@ -0,0 +1,394 @@
...
+https://${account.namespace}/authorize?
+    scope=SCOPE&
+    response_type=code&
+    client_id=${account.clientId}&
+    code_challenge=CODE_CHALLENGE&
+    code_challenge_method=S256&
+    redirect_uri=${account.namespace}/mobile
+```
+
+Please see [this page](/api-auth/tutorials/authorization-code-grant-pkce#3-get-the-user-s-authorization) for detailed information on the User Authorization request parameters.
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: Mention that since in this scenario the app needs only to authenticate the user, not to access an API, they do not need to set the `audience` parameter.

File: articles/client-auth/v2/mobile-desktop.md [hunk 24]
```diff
@@ -0,0 +1,394 @@
...
+
+::: panel-danger Warning
+The Authorization Code flow with PKCE can only be used for Native Clients.
+:::
+
+![](/media/articles/client-auth/mobile-desktop/create-client.png)
+
+Once Auth0 creates the Client, navigate to the Client's **Settings** tab to add the following to the **Allowed Callback URLs** field: `https://${account.namespace}/mobile`
+
+Scroll to the bottom of the page and click **Save**.
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: Add also that they should enable the OIDC-conformant flag at their client (settings > advanced settings > OAuth > OIDC Conformant)

File: articles/client-auth/v2/mobile-desktop.md [hunk 25]
```diff
@@ -0,0 +1,394 @@
...
+Please see [this page](/api-auth/tutorials/authorization-code-grant-pkce#3-get-the-user-s-authorization) for detailed information on the User Authorization request parameters.
+
+::: panel-info Arbitrary Claims
+To improve Client application compatibility, Auth0 returns profile information using an [OIDC-defined structured claim format](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims). This means that arbitrary claims to ID or access tokens must conform to a namespaced format to avoid collisions with standard OIDC claims. For example, if your namespace is `https://foo.com/` and you want to add an arbitrary claim named `myclaim`, you would name the claim `https://foo.com/myclaim`, not `myclaim`.
+:::
+
+As an example, your HTML snippet for your authorization URL might look as follows:
+
+```html
+<a href="https://${account.namespace}/authorize?
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: this example needs some changes:
- remove `audience` param, we only set this to access an API (and in this case we are only authenticating the user)
- update `scope` to include only user information scopes, for example `scope=openid%20profile%20email`

File: articles/client-auth/v2/mobile-desktop.md [hunk 26]
```diff
@@ -0,0 +1,394 @@
...
+  "access_token":"eyJz93a...k4laUWw",
+  "refresh_token":"GEbRxBN...edjnXbL",
+  "id_token":"eyJ0XAi...4faeEoQ",
+  "token_type":"Bearer",
+  "expires_in":86400
+}
+```
+
+:::panel-info Access Tokens
+You can use `access_token` to call the [Authentication API's `/userinfo` endpoint](/api/authentication#get-user-info).
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: "You can use the `access_token`"

File: articles/client-auth/v2/mobile-desktop.md (hunk 17 above)
Comment (mpaktiti) [reply articles/client-auth/v2/mobile-desktop.md]: Can you remove this paragraph? Doesn't really fit it. If we wanted to elaborate further we should show how to validate and decode the token, but I don't think it's needed here. Maybe we can iterate on it at some later point.

File: articles/client-auth/v2/mobile-desktop.md [hunk 27]
```diff
@@ -0,0 +1,394 @@
...
+https://${account.namespace}/authorize?
+    scope=openid%20name%20picture&
+    response_type=code&
+    client_id=${account.clientId}&
+    code_challenge=CODE_CHALLENGE&
+    code_challenge_method=S256&
+    redirect_uri=${account.namespace}/mobile
+```
+
+After the user provides submits their credentials, your app receives an HTTP 302 response with a URL containing the authorization code at the end: `https://YOUR_APP/callback?code=AUTHORIZATION_CODE`
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: Change the `https://YOUR_APP/callback?code=AUTHORIZATION_CODE` to `${account.namespace}/mobile?code=AUTHORIZATION_CODE`

File: articles/client-auth/v2/mobile-desktop.md [hunk 28]
```diff
@@ -0,0 +1,394 @@
...
+    code_challenge=CODE_CHALLENGE&
+    code_challenge_method=S256&
+    redirect_uri=https://${account.namespace}/mobile&
+    connection=github
+```
+
+After the user authenticates, GitHub redirects them back to the specified `redirect_uri` with the `id_token` and `token_type` passed as parameters in the included hash fragment.
+
+```text
+https://YOUR_APP/callback
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: Change the URL to `${account.namespace}/mobile`

File: articles/client-auth/v2/mobile-desktop.md [hunk 29]
```diff
@@ -0,0 +1,394 @@
...
+    scope=openid%20name%20picture&
+    response_type=code&
+    client_id=${account.clientId}&
+    code_challenge=CODE_CHALLENGE&
+    code_challenge_method=S256&
+    redirect_uri=https://${account.namespace}/mobile&
+    connection=github
+```
+
+After the user authenticates, GitHub redirects them back to the specified `redirect_uri` with the `id_token` and `token_type` passed as parameters in the included hash fragment.
```
Comment (mpaktiti) [inline articles/client-auth/v2/mobile-desktop.md]: This is not correct. The response is the authorization code (exactly like in the previous example).

File: articles/tokens/id-token.md [hunk 30]
```diff
@@ -7,7 +7,9 @@ description: How to obtain, use and renew an id_token.
 
 ## Overview
 
-The ID token, usually referred to in our docs as `id_token`, is a [JSON Web Token (JWT)](/jwt) that contains user profile information (like the user's name, email, and so forth), represented in the form of _claims_. These claims are statements about the user, which can be trusted if the consumer of the token can [verify its signature](#validate-an-id-token).
+Auth0's `id_token` is a [JSON Web Token (JWT)](/jwt) of type **Bearer** containing information about the user (such as name and email address) represented in the form of *claims*. These claims are statements about the user, which can be trusted if the consumer of the token can [verify its signature](#validate-an-id-token).
```
Comment (mpaktiti) [inline articles/tokens/id-token.md]: The `access_token` is the `Bearer` token, not the `id_token`. Can you undo this change?

**Review by mpaktiti — CHANGES_REQUESTED** [review]
Hey Katie, left some comments for some minor changes. You can also rename `v2` to `oidc` and then add the document versioning functionality. You can see an example here https://github.com/auth0/docs/pull/3758/files In the p2 docs case we will name the folders `legacy` (the existing doc), and `oidc` (the new docs you write, like this one).

**Comment by mpaktiti** [comment]
Hey @khorne3 , can you please do the following:
- rebase to resolve conflicts
- name versions `legacy` and `current` (like this one https://github.com/auth0/docs/pull/4293 )
- make `current` the default version

You shouldn't need redirects https://github.com/auth0/docs/pull/3754/files#diff-53dc15d3e925b8193de1306c630cca86R1144

Thanks!

**Comment by khorne3** [comment]
@mpaktiti This is ready. Thanks!
